v4/newrelic/application.go

```diff
@@ -76,6 +106,11 @@ func (app *Application) RecordCustomMetric(name string, value float64) {}
 // connection to the Trace Observer is made, a fatal error is reached, or the
 // timeout is hit.
 func (app *Application) WaitForConnection(timeout time.Duration) error {
+	if app == nil {
+		return nil
+	}
+	app.initLogger()
```

Comment:
We should just put this in `NewApplication` so we don't need to worry about it here.